### PR TITLE
Dependencies: Bump prettier dependency, cleanup config

### PR DESCRIPTION
**Pull request by IanVS**
Issue:  N/A

I'm doing a little cleanup of linting/prettier, and found a few issues:

- We had duplicate prettier configs, `.prettierrc` and `prettier.config.js`.  When running prettier via eslint, it seems that `prettier.config.js` was being used, but running on the command line (e.g. with `yarn lint:other`), then `.prettierrc` was used. 
- in our `dependencies` for prettier for a few packages, we were pinning to `<=2.3.0`, due to IE support (https://github.com/storybookjs/storybook/pull/17239).  
- ~`@storybook/components` was setting prettier as a dev dependency, even though it is used as a direct dependency.~ This is necessary, for prettier to be pre-bundled. 

## What I did

- Removing the rc file exposed that html was not formatting correctly with the babel parser, so I removed the override and `yarn lint:other` now passes.
- Since we've dropped IE, I think we can remove the restriction on prettier version, so I did.  This has a potential to save a lot of install size, since older versions of prettier are about 20mb on disk (https://packagephobia.com/result?p=prettier).
- ~I moved prettier to `dependencies` of `@storybook/components`.~ Reverted this, needs to get pre-bundled.

## How to test

CI should pass

### CODE REVIEW

**Review by JReinhold — APPROVED** [review]
LGTM 👍

**Comment by IanVS** [comment]
I reverted the move of prettier from devDeps to deps.  It's back to `devDependencies`.  This is ready for review.

File: code/lib/codemod/package.json [hunk 1]
```diff
@@ -46,7 +46,7 @@
     "globby": "^11.0.2",
     "jscodeshift": "^0.13.1",
     "lodash": "^4.17.21",
-    "prettier": ">=2.2.1 <=2.3.0",
+    "prettier": ">=2.2.1 <=3.0.0",
```
Comment (ndelangen) [inline code/lib/codemod/package.json:49]: Why not simply upgrade to the latest?

Is 3.0.0 a no-go? if so, why?

Comment (IanVS) [reply code/lib/codemod/package.json:49]: 3.0 is making breaking changes, and is only in alpha stage right now.  I thought maybe it's worth waiting to see how it shakes out for a bit.  I'd have to look exactly how we use prettier to see if we're impacted.

**Pull request merged**